pinot-core/src/main/java/com/linkedin/pinot/core/indexsegment/generator/SegmentGeneratorConfig.java

```diff
@@ -144,6 +147,44 @@ public SegmentGeneratorConfig(SegmentGeneratorConfig config) {
     _onHeap = config._onHeap;
   }
 
+  public SegmentGeneratorConfig(TableConfig tableConfig, Schema schema) {
+    Preconditions.checkNotNull(schema);
+    _schema = schema;
+
+    if (tableConfig != null) {
+      IndexingConfig indexingConfig = tableConfig.getIndexingConfig();
+      if (indexingConfig != null) {
+        List<String> noDictionaryColumns = tableConfig.getIndexingConfig().getNoDictionaryColumns();
```

Comment:
You already have `indexingConfig` variable. No need to fetch it from the `tableConfig` again.